test/tftp/tftp_test.rb

```diff
@@ -57,6 +57,40 @@ def test_choose_protocol_and_fetch_wget
     end
   end
 
+  def test_choose_protocol_and_fetch_wget_with_timeouts
+    src = "https://proxy.test"
+    dst = "/destination"
+    tftp_read_timeout = "1000"
+    tftp_connect_timeout = "40"
+    tftp_dns_timeout = "14300"
+
+    Proxy::TFTP::Plugin.settings.expects(:tftp_read_timeout).returns(tftp_read_timeout)
+    Proxy::TFTP::Plugin.settings.expects(:tftp_connect_timeout).returns(tftp_connect_timeout)
+    Proxy::TFTP::Plugin.settings.expects(:tftp_dns_timeout).returns(tftp_dns_timeout)
+
+    start = stub(:start => true)
```

Comment:
This is a bit weird -- the return of #start isn't being used in the code or test, I think it can be safely ignored altogether.

Reply:
That mock is to avoid
```
Error: test_choose_protocol_and_fetch_wget_with_read_timeout(TftpTest): NoMethodError: undefined method `start' for nil:NilClass
```